### PR TITLE
Use zen_nav on Publication and Article pages

### LINKED ISSUE #5781: Navigation: missing menu on viewports
Navigation: missing menu on viewports > 990px (should be similar to blog nav found here: https://foundation.mozilla.org/en/blog/your-holiday-gift-spying-you-find-out/)

We may need to look into enabling "zen mode" to make the hamburger show up the same way as it does in the blog. 

Criteria:
- [ ] Use same navigation as the blog 
- [ ] Ensure the hamburger menu shows up the same way the blog displays it

### PR DESCRIPTION
**Pull request by KalobTaulien**
Closes #5781 

Criteria:
- [x] Use same navigation as the blog 
- [x] Ensure the hamburger menu shows up the same way the blog displays it

**Note**: Instead of editing the templates we're simply telling PublicationPage and ArticlePage to have a "page.zen_nav" property. We also need to add `menu_items` into the page context so the zen nav has something to show.

### CODE REVIEW

File: network-api/networkapi/wagtailpages/pagemodels/publications/article.py [hunk 1]
```diff
@@ -88,9 +89,15 @@ def breadcrumb_list(self):
         """
         return Page.objects.ancestor_of(self).type(PublicationPage).live()
 
+    @property
+    def zen_nav(self):
```
Comment (KalobTaulien) [inline network-api/networkapi/wagtailpages/pagemodels/publications/article.py:93]: Adds an overwritable property to "trick" the templates into thinking this page has "zen_nav" as a property.

File: network-api/networkapi/wagtailpages/pagemodels/publications/publication.py [hunk 2]
```diff
@@ -10,8 +10,9 @@
 from wagtail.images.edit_handlers import ImageChooserPanel
 from wagtail.snippets.edit_handlers import SnippetChooserPanel
 
-from ..mixin.foundation_metadata import FoundationMetadataPageMixin
 from networkapi.wagtailpages.models import BlogAuthor
+from networkapi.wagtailpages.pagemodels.base import Homepage
+from ..mixin.foundation_metadata import FoundationMetadataPageMixin
```
Comment (KalobTaulien) [inline network-api/networkapi/wagtailpages/pagemodels/publications/publication.py:15]: I just re-ordered this as a "best practice" :P

File: network-api/networkapi/wagtailpages/pagemodels/publications/publication.py [hunk 3]
```diff
@@ -110,12 +111,21 @@ def is_chapter_page(self):
...
     def breadcrumb_list(self):
         """
         Get all the parent PublicationPages and return a QuerySet
         """
         return Page.objects.ancestor_of(self).type(PublicationPage).live()
 
+    def get_context(self, request, *args, **kwargs):
+        context = super().get_context(request, *args, **kwargs)
+        # menu_items is required for zen_nav in the templates
+        context['menu_items'] = Homepage.objects.first().get_children().live().in_menu()
```
Comment (KalobTaulien) [inline network-api/networkapi/wagtailpages/pagemodels/publications/publication.py:127]: Adds "menu_items" into the page context so the zen_nav has menu items to show.

Comment (mmmavis) [reply network-api/networkapi/wagtailpages/pagemodels/publications/publication.py:127]: Is this something we can utilize here? https://github.com/mozilla/foundation.mozilla.org/blob/master/network-api/networkapi/wagtailpages/utils.py#L32-L46

e.g., 
```py
def get_context(self, request):
        context = super().get_context(request)
        return set_main_site_nav_information(self, context, 'Homepage')
```

File: network-api/networkapi/wagtailpages/pagemodels/publications/publication.py [hunk 4]
```diff
@@ -110,12 +111,21 @@ def is_chapter_page(self):
         parent = self.get_parent().specific
         return parent.__class__ is PublicationPage
 
+    @property
+    def zen_nav(self):
+        return True
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/pagemodels/publications/publication.py:116]: Q: How is this different than just setting `zen_nav = True` <--- which is what we've been doing

File: network-api/networkapi/wagtailpages/pagemodels/publications/article.py [hunk 5]
```diff
@@ -88,9 +89,15 @@ def breadcrumb_list(self):
...
+        return True
+
     def get_context(self, request, *args, **kwargs):
         context = super().get_context(request, *args, **kwargs)
         # Add get_titles to the page context. This is in get_context() because
         # we need access to the `request` object
         context['get_titles'] = get_richtext_titles(request, self.body, "content")
+        # menu_items is required for zen_nav in the templates
+        context['menu_items'] = Homepage.objects.first().get_children().live().in_menu()
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/pagemodels/publications/article.py:102]: see my other comment

File: network-api/networkapi/wagtailpages/pagemodels/publications/article.py (hunk 1 above)
Comment (mmmavis) [reply network-api/networkapi/wagtailpages/pagemodels/publications/article.py:93]: see my other comment

File: network-api/networkapi/wagtailpages/pagemodels/publications/publication.py (hunk 4 above)
Comment (KalobTaulien) [reply network-api/networkapi/wagtailpages/pagemodels/publications/publication.py:116]: Currently, nothing. It's just overwrite-able through class inheritance. Do we prefer `zen_nv = True` as a concrete property?

Comment (mmmavis) [reply network-api/networkapi/wagtailpages/pagemodels/publications/publication.py:116]: No pref here. Marking this as resolved!

**Review by mmmavis — APPROVED** [review]
There's one line of inline comment I think we should remove. Once that's removed, feel free to merge this PR!

**Pull request merged**